File: examples-sdk/kotlin/033-basic-shm-write-through/main.kt
// 033-basic-shm-write-through — Kotlin SDK equivalent
// Compile: vil compile --from kotlin --input 033-basic-shm-write-through/main.kt --release

fun main() {
    val server = VilServer("realtime-analytics-dashboard", 8080)
    val catalog = ServiceProcess("catalog")
    catalog.endpoint("POST", "/catalog/search", "catalog_search")
    catalog.endpoint("GET", "/catalog/health", "catalog_health")
    server.service(catalog)
    server.compile()
}

File: examples-sdk/swift/033-basic-shm-write-through/main.swift
// 033-basic-shm-write-through — Swift SDK equivalent
// Compile: vil compile --from swift --input 033-basic-shm-write-through/main.swift --release

let server = VilServer(name: "realtime-analytics-dashboard", port: 8080)
let catalog = ServiceProcess(name: "catalog")
catalog.endpoint(method: "POST", path: "/catalog/search", handler: "catalog_search")
catalog.endpoint(method: "GET", path: "/catalog/health", handler: "catalog_health")
server.service(catalog)
server.compile()
